File: hotspot_analysis/social-cards/2026-06-08_LLM_Agent_Coding_Evolution/render.mjs
import { chromium } from 'playwright-core';
import { join, dirname } from 'path';
import { mkdirSync } from 'fs';
import { fileURLToPath } from 'url';

const __dirname = dirname(fileURLToPath(import.meta.url));
const HTML = join(__dirname, 'index.html');
const OUT = join(__dirname, 'output');
mkdirSync(OUT, { recursive: true });

const targets = [
  ['#wechat-21x9',   'wechat-21x9-cover.png'],
  ['#wechat-1x1',    'wechat-1x1-cover.png'],
  ['#wechat-pair-preview', 'wechat-cover-pair-preview.png'],
];

const browser = await chromium.launch({
  executablePath: '/opt/microsoft/msedge/msedge',
  headless: true,
  args: ['--no-sandbox', '--disable-gpu', '--font-render-hinting=none'],
});

const page = await browser.newPage({ viewport: { width: 3800, height: 2000 } });
await page.goto(`file://${HTML}`, { waitUntil: 'domcontentloaded', timeout: 60000 });
await page.waitForTimeout(5000);

for (const [sel, name] of targets) {
  const el = page.locator(sel);
  await el.screenshot({ path: join(OUT, name), type: 'png' });
  console.log(`✓ ${name}`);
}

await browser.close();
console.log('Done — all images in output/');

File: hotspot_analysis/social-cards/2026-06-25_labubu-emotional-economy/render.py
# /// script
# requires-python = ">=3.10"
# dependencies = ["playwright"]
# ///
"""Render WeChat cover pair using Playwright via uv."""
import asyncio
from pathlib import Path
from playwright.async_api import async_playwright

TASK = Path(__file__).parent
HTML = TASK / "index.html"
OUT = TASK / "output"
OUT.mkdir(exist_ok=True)

TARGETS = [
    ("#wechat-21x9", "wechat-21x9.png", 2400, 900),
    ("#wechat-1x1", "wechat-1x1.png", 1080, 1080),
]

async def main():
    async with async_playwright() as pw:
        browser = await pw.chromium.launch(
            executable_path="/opt/microsoft/msedge/msedge",
            headless=True,
            args=["--no-sandbox", "--disable-gpu", "--font-render-hinting=none"],
        )
        page = await browser.new_page()
        await page.set_viewport_size({"width": 2400, "height": 15000})
        await page.goto(f"file://{HTML.resolve()}", wait_until="domcontentloaded", timeout=60000)
        await page.wait_for_timeout(5000)

        for selector, name, w, h in TARGETS:
            el = page.locator(selector)
            await el.screenshot(path=str(OUT / name), type="png")
            print(f"✓ {name} ({w}×{h})")

        await browser.close()
        print(f"Done: {OUT}")

asyncio.run(main())
